### ISSUE TITLE
Flight plan drawn incorrectly on ND

### ISSUE THREAD

**Issue by ntlgr** [flybywiresim/a32nx#6776]
### Aircraft Version

Development

### Build info

```json
{
  "built": "2022-02-21T07:56:45+00:00",
  "ref": "master",
  "sha": "29ebf866b086c2be9331241cbd5fb8858d6f7f95",
  "actor": "sidnov",
  "event_name": "manual",
  "pretty_release_name": "master:29ebf866"
}
```

### Describe the bug

The flight plan is drawn incorrectly on ND. Some strange turns and a non-existent flyback is drawn in otherwise correctly sequenced waypoints. In the provided screenshot, the flight plan appears on the FMC and at the same time the problem is seen the ND.
![Microsoft Flight Simulator 18_02_2022 13_29_31](https://user-images.githubusercontent.com/41379877/155015333-c058295b-e093-4910-aaad-5d4208029c6b.jpg)

### Expected behavior

The flight plan should appear as a green line connecting the upcoming waypoints using the shortest flyable path between them.

### Steps to reproduce

Not specific to a single flightplan; has been observed several times especially when approaching the IAF for a given flight. 
Screenshot belongs to a simbrief-generated flight plan from MMUN to MMMX.

**Comment by ntlgr** [comment]
Routing was: VOBE2A VOBED V56 CZA UT20 MEDOS UT17 USBER DCT AVALI DCT REXES
Specifically waypoints: MMUN CUN D258G VOBED UDGUV CZA ALPOM MEDOS USBER AVALI REXES MMMX

**Comment by ntlgr** [comment]
I can attach the simbrief file?

**Comment by ntlgr** [comment]
You are right about CZA, it shouldn't be there the sequence was incorrect. But was imported from simbrief!

**Comment by Benjozork** [comment]
CZA should be earlier in the flight plan. This is likely a bug with simBrief importing.

Can you post your simBrief ID actually ? It will allow me to use your OFP in the aircraft. Make sure to not generate any OFPs for about a day so I can load it myself.

**Comment by ntlgr** [comment]
Could be. Simbrief alias: MikePapa
I'm afraid I have already generated another one :((

**Comment by Benjozork** [comment]
alright. for future ref, this is your username, not your ID. ID will be automatically indicated in the flyPad AOC settings

**Comment by Benjozork** [comment]
if you have the issue with another plan, just reply here

**Comment by ntlgr** [comment]
Will do. Thanks for looking into it. Probably a non issue

**Comment by Emenuez** [comment]
It usually happens on some occasions. What I do is do is make a DIR in the FMC the next waypoints and that usually solves the problem. It's not ideal but it works.

**Comment by merlinblack** [comment]
I had a weird one when flying from Beirut (OLBA) to Izmir (LTBJ) - imported from Simbrief. The STAR at Izmir was messed up. I'll try again and report back.

**Comment by Benjozork** [comment]
Need screenshots.

**Comment by kilocharlie1** [comment]
I am having the same issue with my ND, except it doesn't draw the line at all in correspondence with the waypoints, rather straight out from the runway I took off from

**Comment by merlinblack** [comment]
Experimental is crashing on me at the moment when I 'Request ...' from the MCDU to download from simbrief, so I can't try it again right now. My Simbreif pilot id is 392990 and the flight plan is my OFP - anyone wants to try it.

**Comment by ntlgr** [comment]
@Benjozork I believe I have found what the issue is. I think the turn arcs on the ND are drawn based on the current aircraft speed, and not based on the planned speed for that leg,
Some evidence: If you look at the two screenshots I have attached here, you will see the ILS Z approach to SEQM 36. The waypoints are configured manually (as the approach is not available in the data set). While I am still enroute, the flight path is drawn erratically. Later, I descend and decelerate to the speeds required to fly the approach, and the ND is drawing it perfectly. The plane also flew the path correctly. NO changes have been made to the waypoints in between the screenshots.
Hope this helps
![cap1](https://user-images.githubusercontent.com/41379877/155547224-e03c6700-cb6b-4b7a-ad41-942c28c35aa6.jpg)
![cap2](https://user-images.githubusercontent.com/41379877/155547284-3b49becd-87a5-4974-a7de-c146aa4a6110.jpg)

**Comment by Benjozork** [comment]
That is not the same issue, though.
